### PR TITLE
Unify the definition of GCCVER and remove duplicate code.

### PR DESCRIPTION
**Pull request by cuiziwei1**

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



I need the content of the PR to assess if it meets the NuttX requirements. Please provide the following:

* **The actual PR content:** This includes the PR description, code changes, and any related files.
* **Links to the PR:** If the PR is already open on GitHub, please share the link. 

With this information, I can review the PR against the NuttX requirements and provide a concise assessment.

File: tools/Config.mk [hunk 1]
```diff
@@ -732,6 +732,9 @@ export DEFINE_PREFIX ?= $(subst X,,${shell $(DEFINE) "$(CC)" X 2> ${EMPTYFILE}})
 export INCDIR_PREFIX ?= $(subst "X",,${shell $(INCDIR) "$(CC)" X 2> ${EMPTYFILE}})
 export INCSYSDIR_PREFIX ?= $(subst "X",,${shell $(INCDIR) -s "$(CC)" X 2> ${EMPTYFILE}})
 
+# Get the GCC version number
+export GCCVER = $(shell $(CXX) --version | grep ++ | sed -E 's/.* ([0-9]+\.[0-9]+).*/\1/' | cut -d'.' -f1)
```
Comment (xuxin930) [inline tools/Config.mk:736]:
```suggestion
ifeq ($(CONFIG_ARCH_TOOLCHAIN_GNU), y)
  export GCCVER = $(shell $(CC) --version | grep ++ | sed -E 's/.* ([0-9]+\.[0-9]+).*/\1/' | cut -d'.' -f1)
endif
```

File: arch/arm/src/cmake/gcc.cmake [hunk 2]
```diff
@@ -62,7 +62,7 @@ endif()
 # array subscript [0] is outside array bounds:
 # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105523
 
-execute_process(COMMAND ${CMAKE_C_COMPILER} --version
+execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version
```
Comment (xuxin930) [inline arch/arm/src/cmake/gcc.cmake:65]:
```suggestion
execute_process(COMMAND ${CMAKE_C_COMPILER} --version
                OUTPUT_VARIABLE GCC_VERSION_OUTPUT)
string(REGEX MATCH "\\+\\+.* ([0-9]+)\\.[0-9]+" GCC_VERSION_REGEX
             "${GCC_VERSION_OUTPUT}")
set(GCCVER ${CMAKE_MATCH_1} PARENT_SCOPE)
```

**Comment by cuiziwei1** [comment]
@anchao @xiaoxiang781216 please review this patch to resolve https://github.com/apache/nuttx/pull/14130#issuecomment-2408282192

**Comment by xuxin930** [comment]
hi @anchao please review whether this location is reasonable.

**Review by anchao — APPROVED** [review]

**Review by GUIDINGLI — APPROVED** [review]
LGTM

File: arch/arm/src/cmake/gcc.cmake [hunk 3]
```diff
@@ -66,7 +66,9 @@ execute_process(COMMAND ${CMAKE_C_COMPILER} --version
                 OUTPUT_VARIABLE GCC_VERSION_OUTPUT)
 string(REGEX MATCH "\\+\\+.* ([0-9]+)\\.[0-9]+" GCC_VERSION_REGEX
              "${GCC_VERSION_OUTPUT}")
-set(GCCVER ${CMAKE_MATCH_1})
+set(GCCVER
+    ${CMAKE_MATCH_1}
+    PARENT_SCOPE)
 
 if(GCCVER EQUAL 12)
```
Comment (pkarashchenko) [inline arch/arm/src/cmake/gcc.cmake:73]: Should this also be greater equal?

File: arch/arm/src/common/Toolchain.defs [hunk 4]
```diff
@@ -309,10 +309,6 @@ else
    # Wrong warning array subscript [0] is outside array bounds:
    # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105523
 
-   ifeq ($(GCCVER),)
-     export GCCVER := $(shell $(CC) --version | grep gcc | sed -E "s/.* ([0-9]+\.[0-9]+).*/\1/" | cut -d'.' -f1)
-   endif
-
    ifeq ($(GCCVER),12)
```
Comment (pkarashchenko) [inline arch/arm/src/common/Toolchain.defs:312]: Should this be `($(shell expr $(GCCVER) \>= 12), 1)`?

File: libs/libxx/libcxx.cmake [hunk 5]
```diff
@@ -95,14 +95,6 @@ list(APPEND SRCS ${SRCSTMP})
...
 
-if(NOT DEFINED GCCVER)
-  execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version
-                  OUTPUT_VARIABLE GCC_VERSION_OUTPUT)
-  string(REGEX MATCH "\\+\\+.* ([0-9]+)\\.[0-9]+" GCC_VERSION_REGEX
-               "${GCC_VERSION_OUTPUT}")
-  set(GCCVER ${CMAKE_MATCH_1})
-endif()
-
 if(GCCVER EQUAL 12)
```
Comment (pkarashchenko) [inline libs/libxx/libcxx.cmake:98]: Should this be greater equal?

File: arch/arm/src/cmake/gcc.cmake (hunk 3 above)
Comment (cuiziwei1) [reply arch/arm/src/cmake/gcc.cmake:73]: ok, please review https://github.com/apache/nuttx/pull/14241

**Pull request merged**